features/events/daemon/src/main/java/org/opennms/netmgt/collection/EventListenerCollector.java

```diff
@@ -0,0 +1,193 @@
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.collection;
+
+import org.opennms.netmgt.collection.api.*;
+import org.opennms.netmgt.collection.core.DefaultCollectionAgent;
+import org.opennms.netmgt.collection.support.AbstractCollectionAttribute;
+import org.opennms.netmgt.collection.support.AbstractCollectionAttributeType;
+import org.opennms.netmgt.collection.support.AbstractCollectionResource;
+import org.opennms.netmgt.collection.support.SingleResourceCollectionSet;
+import org.opennms.netmgt.collection.support.builder.*;
+import org.opennms.netmgt.config.DataCollectionConfigFactory;
+import org.opennms.netmgt.config.api.EventConfDao;
+import org.opennms.netmgt.dao.api.IpInterfaceDao;
+import org.opennms.netmgt.events.api.EventListener;
+import org.opennms.netmgt.events.api.EventSubscriptionService;
+import org.opennms.netmgt.events.api.model.IEvent;
+import org.opennms.netmgt.events.api.model.IParm;
+import org.opennms.netmgt.model.OnmsIpInterface;
+import org.opennms.netmgt.model.ResourceTypeUtils;
+import org.opennms.netmgt.rrd.RrdRepository;
+import org.opennms.netmgt.xml.eventconf.Collection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.PlatformTransactionManager;
+
+import javax.annotation.PostConstruct;
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This collector will convert event into time series data. It depends on eventconf.xsd's collection tag.
+ */
+public class EventListenerCollector implements EventListener {
+    private static final Logger LOG = LoggerFactory.getLogger(EventListenerCollector.class);
+
+    @Autowired
+    private EventConfDao eventConfDao;
+
+    @Autowired
+    private EventSubscriptionService eventSubscriptionService;
+
+    @Autowired
+    private PersisterFactory persisterFactory;
+
+    @Autowired
+    private IpInterfaceDao ipInterfaceDao;
+
+    @Autowired
+    private PlatformTransactionManager platformTransactionManager;
+
+    @PostConstruct
+    public void init() {
+        eventSubscriptionService.addEventListener(this);
+    }
+
+    @Override
+    public String getName() {
+        return this.getClass().getSimpleName();
+    }
+
+    @Override
+    public void onEvent(IEvent e) {
+        var eventconf = eventConfDao.findByUei(e.getUei());
+        if (eventconf == null || eventconf.getCollections().isEmpty()) {
+            return;
+        }
+        List<Collection> collections = eventconf.getCollections();
+
+        OnmsIpInterface iface = ipInterfaceDao.findByNodeIdAndIpAddress(e.getNodeid().intValue(),
+                e.getInterfaceAddress().getHostAddress());
+        CollectionAgent agent = DefaultCollectionAgent.create(iface.getId(), ipInterfaceDao, platformTransactionManager);
```

Comment:
Can replace this with `CollectionAgentFactory` interface

Reply:
replaced